src/main/java/no/ndla/taxonomy/service/dtos/TaxonomyContextDTOFactory.java

```diff
@@ -0,0 +1,99 @@
+package no.ndla.taxonomy.service.dtos;
+
+import no.ndla.taxonomy.domain.*;
+import no.ndla.taxonomy.repositories.RelevanceRepository;
+import no.ndla.taxonomy.rest.v1.dtos.nodes.searchapi.*;
+import org.springframework.stereotype.Service;
+
+import java.net.URI;
+import java.util.*;
+
+@Service
+public class TaxonomyContextDTOFactory {
+    private final RelevanceRepository relevanceRepository;
+
+    public TaxonomyContextDTOFactory(RelevanceRepository relevanceRepository) {
+        this.relevanceRepository = relevanceRepository;
+    }
+
+    public List<TaxonomyContextDTO> fromNodes(List<Node> nodes, boolean filterVisibles) {
+        return nodes.stream().flatMap(node -> {
+            return node.buildPaths().stream().map(np -> {
+                return fromNodeAndPath(node, np, filterVisibles);
+            }).filter(Optional::isPresent).map(Optional::get);
+        }).toList();
+    }
+
+    private Optional<TaxonomyContextDTO> fromNodeAndPath(Node node, NodePath nodePath, boolean filterVisibles) {
+        if (filterVisibles && !nodePath.isVisible())
+            return Optional.empty();
+
+        var maybeSubjectNode = nodePath.getNodes().stream().findFirst()
+                .filter(s -> s.getPublicId() != node.getPublicId());
+        if (maybeSubjectNode.isEmpty())
+            return Optional.empty();
+
+        var subjectNode = maybeSubjectNode.get();
+        var subjectNames = LanguageField.nameFromNode(subjectNode);
+        var path = nodePath.toString();
+
+        var relevance = nodePath.getBaseRelevance().orElseGet(() -> {
+            var coreId = URI.create("urn:relevance:core");
+            return relevanceRepository.findFirstByPublicIdIncludingTranslations(coreId).orElseThrow();
+        });
+
+        var relevanceTranslations = new LanguageField<String>();
+        relevanceTranslations.put(DefaultLanguage, relevance.getName());
+        relevance.getTranslations()
+                .forEach(trans -> relevanceTranslations.put(trans.getLanguageCode(), trans.getName()));
+
+        var relevanceId = nodePath.getBaseRelevance().map(DomainEntity::getPublicId)
+                .orElse(URI.create("urn:relevance:core"));
```

Comment:
Vil ikkje dette være det samme? Du har allerede et relevanceobjekt.
```suggestion
        var relevanceId = relevance.getPublicId();
```